### PR TITLE
fix(ui): tag api related changes to UI

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes :

I worked on the 
- Tag Category --> Classification
- Create / Update / Delete / Get for classification should work properly
- Tags should be listed for tags dropdown
- Tags should be listed within category properly

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
Screenshot 2022-12-27 at 8 06 44 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx [hunk 1]
```diff
@@ -432,13 +432,14 @@ const DashboardDetails = ({
...
         if (
           values[0].status === SettledStatus.FULFILLED &&
           values[0].value.data
         ) {
-          tagsAndTerms = getTaglist(values[0].value.data).map((tag) => {
+          const tagList = await getTaglist(values[0].value.data);
+          tagsAndTerms = tagList.map((tag) => {
             return { fqn: tag, source: 'Tag' };
           });
         }
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx:445]: same here, async-await with .then and .catch

File: openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx [hunk 2]
```diff
@@ -104,15 +104,23 @@ const EntityTable = ({
...
+
+          tagsAndTerms =
+            tagList.length !== 0
+              ? tagList.map((tag) => {
+                  return {
+                    fqn: tag,
+                    source: 'Tag',
+                  };
+                })
+              : [];
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx:123]: same here

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryDetails/GlossaryDetails.component.tsx [hunk 3]
```diff
@@ -196,8 +196,9 @@ const GlossaryDetails = ({ permissions, glossary, updateGlossary }: props) => {
   const fetchTags = () => {
     setIsTagLoading(true);
     getClassifications()
-      .then((res) => {
-        setTagList(getTaglist(res.data));
+      .then(async (res) => {
+        const tagList = await getTaglist(res.data);
+        setTagList(tagList);
       })
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryDetails/GlossaryDetails.component.tsx:202]: same here

File: openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermsV1.component.tsx [hunk 4]
```diff
@@ -172,8 +172,9 @@ const GlossaryTermsV1 = ({
   const fetchTags = () => {
     setIsTagLoading(true);
     getClassifications()
-      .then((res) => {
-        setTagList(getTaglist(res.data));
+      .then(async (res) => {
+        const tagList = await getTaglist(res.data);
+        setTagList(tagList);
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermsV1.component.tsx:177]: here as well

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx [hunk 5]
```diff
@@ -124,13 +124,14 @@ const MlModelFeaturesList: FC<MlModelFeaturesListProp> = ({
...
-      .then((values) => {
+      .then(async (values) => {
         let tagsAndTerms: TagOption[] = [];
         if (
           values[0].status === SettledStatus.FULFILLED &&
           values[0].value.data
         ) {
-          tagsAndTerms = getTaglist(values[0].value.data).map((tag) => {
+          const tagList = await getTaglist(values[0].value.data);
+          tagsAndTerms = tagList.map((tag) => {
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx:134]: here also

File: openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx [hunk 6]
```diff
@@ -264,13 +264,14 @@ const EntityPageInfo = ({
...
-      .then((values) => {
+      .then(async (values) => {
         let tagsAndTerms: TagOption[] = [];
         if (
           values[0].status === SettledStatus.FULFILLED &&
           values[0].value.data
         ) {
-          tagsAndTerms = getTaglist(values[0].value.data).map((tag) => {
+          const tagList = await getTaglist(values[0].value.data);
+          tagsAndTerms = tagList.map((tag) => {
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:274]: here also

**Review by ShaileshParmar11 — COMMENTED** [review]
Few comments

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx:445]: Above we need to resolve multiple promise together & below we need to prepare `tagsAndTerms` before proceeding further

File: openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx (hunk 2 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityTable/EntityTable.component.tsx:123]: same as above, We can keep note of this and probably update it all together with proper approach

File: openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx (hunk 5 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/MlModelDetail/MlModelFeaturesList.tsx:134]: same as mentioned above

File: openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx (hunk 6 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:274]: same as above

File: openmetadata-ui/src/main/resources/ui/.eslintrc.yaml [hunk 7]
```diff
@@ -167,10 +167,14 @@ rules:
   '@typescript-eslint/explicit-function-return-type':
     - off
 
+  # disable the rule for all files
+  '@typescript-eslint/explicit-module-boundary-types':
+    - off
+
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/.eslintrc.yaml:173]: Why we are disabling this rule?

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/.eslintrc.yaml (hunk 7 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/.eslintrc.yaml:173]: It will print tones of errors. We will slowly enable them if needed

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by ShaileshParmar11 — DISMISSED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**